### PR TITLE
build2 0.17.0

### PR DESCRIPTION
**Pull request by boris-kolpackov**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/b/build2.rb [hunk 1]
```diff
@@ -1,35 +1,19 @@
...
 
-  bottle do
-    sha256 arm64_sonoma:   "cfcd09aa221a08dd5b190e1d471822946c7e08210f74dc3be4ef8f89f088d937"
-    sha256 arm64_ventura:  "fc0c8d37ea3a336f182133daeb135699a45b34d8fa02af56dbc104fff33d6f3c"
-    sha256 arm64_monterey: "93e00c45fcac69a862a26b5e39004c1217675f6a482812107b27d85ff9ab2318"
-    sha256 sonoma:         "aa3e700461c6a0c2052cb3d89a78f9f37bd65e9b8a6fcb9191a5a0b2f3d29540"
-    sha256 ventura:        "61a7a28750af25b609ed325f532773d2624e23ca6ebfb616574cce759f8a4bc0"
-    sha256 monterey:       "1f90f3b72cadd6d429164f3db7f49a200e72da2faa1316a0f113bbe2d264b287"
-    sha256 x86_64_linux:   "92b0ca8b79c117d3e4b0462cf0ab7e1d2d4fffe7c17331e02a7e0b336ac505ab"
-  end
```
Comment (chenrui333) [inline Formula/b/build2.rb:21]: no need to touch it

File: Formula/b/build2.rb [hunk 2]
```diff
@@ -1,35 +1,19 @@
...
-
-  depends_on "pkgconf"
-
   uses_from_macos "curl"
   uses_from_macos "openssl"
-  uses_from_macos "sqlite"
```
Comment (chenrui333) [inline Formula/b/build2.rb:27]: any ref for these dep updates?

File: Formula/b/build2.rb (hunk 1 above)
Comment (boris-kolpackov) [reply Formula/b/build2.rb:21]: Sorry, will add back.

File: Formula/b/build2.rb (hunk 2 above)
Comment (boris-kolpackov) [reply Formula/b/build2.rb:27]: The background/details are here: https://lists.build2.org/archives/users/2024-June/001117.html

Related:

https://github.com/Homebrew/homebrew-core/pull/167330
https://github.com/build2/build2/issues/381

File: Formula/b/build2.rb (hunk 1 above)
Comment (boris-kolpackov) [reply Formula/b/build2.rb]: Done.

**Review by chenrui333 — APPROVED** [review]
Thanks!

**Comment by boris-kolpackov** [comment]
Thanks for a fast review!

**Pull request merged**